datasource/api.js

```diff
@@ -8,97 +8,170 @@
   4) isContractAddr(addr: String): Boolean
 */
 
-import fetch from 'node-fetch'
-import { stripHexPrefix } from '../util.js'
+import fetch from "node-fetch";
+import { stripHexPrefix } from "../util.js";
 
-class Api {
+import zilp from "@zilliqa-js/zilliqa";
+const { Zilliqa } = zilp;
 
-  constructor() {
-    this.networkUrl = 'https://dev-api.zilliqa.com/'
+class Api {
+  constructor(networkUrl = "https://api.zilliqa.com/") {
```

Comment:
We need to abstract this based on environment variables, otherwise when we merge into staging it'll start indexing mainnet chain into our staging database which stores info on the testnet chain

so for example NODE_ENV
if stg use dev-api.zilliqa.com
if prd use api.zilliqa.com

Reply:
oh i see there's a NETWORK_URL environment variable

Reply:
but the api.js is still referencing the constructor's variable instead of the environment variable?

Reply:
The api.js uses mainnet for default, but the constructed API on index.js is referencing the environment var